src/test/java/org/jenkinsci/remoting/protocol/impl/SSLEngineFilterLayerTest.java

```diff
@@ -558,8 +559,8 @@ public void sendingBiggerAndBiggerBatches(NetworkLayerFactory serverFactory, Net
 
         client.awaitByteContent(SequentialSender.matcher(amount));
 
-        client.close();
-        server.close();
+        client.close(null);
```

Comment:
Based on the earlier comments in this pull request, I assumed that
```
client.close(null);
```

might have problems.  Yet it does not. Any idea of the differences between this working case and the earlier hanging case?

Reply:
Probably a different concrete implementation class at runtime. I haven't verified that, yet, though.